### DISCUSSION TITLE
Inspecting Javascript objects returned by the store (like match(....))

### DISCUSSION THREAD

**Discussion by BatPandaPancake** [oxigraph/oxigraph#453] [Q&A]
Hi there, 

I've been playing around with oxigraph and noticed that console log / dir or any attempt to get all the properties of an object like through console log, Object.entries(), the ...spread operator and other Object functions result in an Object{ ptr: <number> } being returned. It seems like the only way to access properties is through specifying what property you want, such as through the dot operator.

I was trying to dig into it and it seems related to JS Proxies / Reflect, but haven't gone further. Is there an accepted way to do this. If not, could this underlying pointer structure be abstracted away with further defining Object / proxy handlers?

**Comment by Tpt** [comment]
Hi! Oxigraph for JavaScript is written in Rust compiled to WebAssembly using [wasm-bindgen](https://github.com/rustwasm/wasm-bindgen). What you see is objects that are just pointers to WASM memory, hence the limitation. I hope to migrate in the future to [reference types](https://rustwasm.github.io/docs/wasm-bindgen/reference/reference-types.html) that should improve that (#455).
